shared/js/background/tab-manager.es6.js

```diff
@@ -71,12 +79,17 @@ class TabManager {
      * later on when webrequests start coming in.
      */
     createOrUpdateTab (id, info) {
-        if (!tabManager.get({'tabId': id})) {
+        const tab = tabManager.get({'tabId': id})
+
+        if (!tab) {
             info.id = id
             tabManager.create(info)
         } else {
-            let tab = tabManager.get({tabId: id})
-            if (tab && info.status) {
+            if (info.url) {
+                tab.url = info.url
+            }
```

Comment:
saving to storage at the end and URL upgrading are only changes here, everything else is just a clean up